backend/mpmediabridge.m

```diff
@@ -0,0 +1,85 @@
+//go:build darwin
+
+#import "mpmediabridge.h"
+
+/**
+ * C bridge registering callbacks for media playback events using the native CommandCenter API.
+ */
+void register_os_remote_commands() {
+    MPRemoteCommandCenter *commandCenter = [MPRemoteCommandCenter sharedCommandCenter];
+    [commandCenter.playCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
+        os_remote_command_callback(PLAY);
+        return MPRemoteCommandHandlerStatusSuccess;
+    }];
+
+    [commandCenter.pauseCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
+        os_remote_command_callback(PAUSE);
+        return MPRemoteCommandHandlerStatusSuccess;
+    }];
+
+    [commandCenter.togglePlayPauseCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
+        os_remote_command_callback(TOGGLE);
+        return MPRemoteCommandHandlerStatusSuccess;
+    }];
+
+    [commandCenter.stopCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
+        os_remote_command_callback(STOP);
+        return MPRemoteCommandHandlerStatusSuccess;
+    }];
+
+    [commandCenter.nextTrackCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
+        os_remote_command_callback(NEXT_TRACK);
+        return MPRemoteCommandHandlerStatusSuccess;
+    }];
+
+    [commandCenter.previousTrackCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(MPRemoteCommandEvent * _Nonnull event) {
```

Comment:
I noticed that in the dock controls (where the album art and metadata is shown) the back button is not available, though it does go back if you hit the back media key. Is there an API to register that the player is able to go back, and the OS should show the button?

Reply:
Do you have a screenshot? The back button is registered (and as a result the media keys work) I’m running ‘Sanoma’ at least there the lack of back button in the ‘small’ widget is as designed by Apple , they decided for whatever reason the back button is not needed there.

There’s a chance this could be different in Ventura

Reply:
Ok, just saw your other comment. It looks like this is the same in Ventura, it’s Apples design and an OS feature so we can’t control the buttons there.

regarding playback duration and scrubbing. It’s for sure possible to implement this, but considerably more involved to keep it in-sync with the player. I think implementation of that should be done as a separate task.

Reply:
I think you are right that it's an Apple decision to not put the back button in the small view - TIL